src/YTable.cc

```diff
@@ -194,6 +194,11 @@ YTable::propertySet()
 	 * @property string	Item		QueryWidget only: Return one complete item
 	 * @property string	IconPath	Base path for icons
 	 * @property bool	MultiSelection	Flag: User can select multiple items (read-only)
+         *
+	 * @property map<ItemID, string> OpenItems 	Map of IDs of all open items (read-only)
+         *                                              to either "ID" or "Text":
+         *                                              { "myFirstItemID": "ID",
+         *                                                "myThirdItem":   "Text" }
```

Comment:
OK, I have tried [the example](https://github.com/yast/yast-ycp-ui-bindings/pull/57/files#diff-93d22226a4b73cfab32c239cce729e0bd2bac087f70817f51caa8349847749d9) and I still have questions:

What if the item tree is nested (the example has only 1 sublevel)? is OpenItems flat or nested? -> flat

In the example all of the keys have the value `"ID"`. What does it mean `"Text"`, or, how do I modify the example to show it?

Reply:
It only reports items as open if they actually have children. See also

  https://github.com/libyui/libyui/blob/master/src/YTreeItem.h#L118

The map value returns "Text" only if the item does not have an ID. But as written before, the value in that map is really pretty meaningless. A map allows constant-time access, so checking it if it contains an ID is theoretically faster than using a list; but that won't ever matter here because we are typically using a small number of items anyway.

Reply:
When I open a second-level item but collapse its parent, the 2nd level item is still listed as open even though it is not visible. Both Qt and Ncurses. Is that intentional?

Reply:
Yes, that's intentional. If you play around in other Qt applications, you will find that they also remember the open sub-branches when you close one higher up. Most (all?) file manager behave the same way.

I am never really sure if I like that behaviour, but it's consistent among similar applications.